common-npm-packages/packaging-common/util.ts

```diff
@@ -102,6 +116,86 @@ export enum LogType {
     error
 }
 
+export function getAccessToken(nugetFeedType:string, endpointInputKey:string, feedInputKey:string, packageToolType: PackageToolType) : string {
+    let accessToken: string = getSystemAccessToken();
+    if (nugetFeedType.toLowerCase() == "internal") {
+        let feed = getProjectAndFeedIdFromInputParam(feedInputKey);
+        accessToken = getUserAccessToken(tl.getInput(endpointInputKey), packageToolType, feed);
+    }
+
+    if (!accessToken) {
+        tl.debug(`Access token not set. Using System Access token.`);
+        return getSystemAccessToken();
+    }
+    return accessToken;
+}
+
+function getUserAccessToken(endpointName: string, packageToolType: PackageToolType, feed: any): string {
+    let token = "";
+    tl.debug("Checking if the token is already set.");
+
+    if(endpointName) {
+        tl.debug(`Checking if the endpoint ${endpointName} provided by user, can be used.`);
+        token = getAccessTokenFromServiceConnectionForInternalFeeds(endpointName);
+    }
+    else {
+        tl.debug("Checking if the credentials are set in the environment.");
+        token = getAccessTokenFromEnvironmentForInternalFeeds(feed, packageToolType);
+    }
+
+    return token;
+}
+
+function getAccessTokenFromServiceConnectionForInternalFeeds(endpointName: string): string {
+    let token: string = "";
+    let auth = tl.getEndpointAuthorization(endpointName, true);
+    let scheme = tl.getEndpointAuthorizationScheme(endpointName, true).toLowerCase();
+    switch(scheme)
+    {
+        case ("token"):
+            token = auth.parameters["apitoken"];
+            break;
+        default:
+            tl.warning("Invalid authentication type for internal feed. Use token based authentication.");
+            break;
+    }
+
+    return token;
+}
+
+function getAccessTokenFromEnvironmentForInternalFeeds(feed: any, packageToolType: PackageToolType) : string {
+    let token: string = "";
+
+    switch(packageToolType)
+    {
+        case PackageToolType.NuGetCommand:
+        case PackageToolType.DotNetCoreCLI:
+            const JsonEndpointsString = process.env["VSS_NUGET_EXTERNAL_FEED_ENDPOINTS"];
```

Comment:
I'm wondering why we chose to target these env variables? If I understand correctly, these are used and set by credential provider. If related to cred provider, we may want to make changes there instead.

Reply:
"VSS_NUGET_EXTERNAL_FEED_ENDPOINTS" is being set by NugetAuthenticate task as well when a service connection is passed. So, here we are reading the value to check if the pipeline has credential set by nuget authenticate. 
There are 2 cases, 
Credential set on the pipeline using NugetAuthenticate task
Credential passed to NugetCommand/DotNetCoreCLI as service connection input.
This code handles the case #1, where credential from service connection is set using NugetAuthenticate task. Here we are reading the access token so that it can be passed to nuget.exe.